apm-agent-plugins/apm-log-shader-plugin/src/main/java/co/elastic/apm/agent/log/shader/AbstractLogShadingHelper.java

```diff
@@ -0,0 +1,102 @@
+/*-
+ * #%L
+ * Elastic APM Java agent
+ * %%
+ * Copyright (C) 2018 - 2020 Elastic and contributors
+ * %%
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * #L%
+ */
+package co.elastic.apm.agent.log.shader;
+
+import co.elastic.apm.agent.configuration.CoreConfiguration;
+import co.elastic.apm.agent.impl.ElasticApmTracer;
+import co.elastic.apm.agent.logging.LoggingConfiguration;
+import com.blogspot.mydailyjava.weaklockfree.WeakConcurrentMap;
+
+import javax.annotation.Nullable;
+
+/**
+ * The abstract Log shading helper- loaded as part of the agent core (agent CL / bootstrap CL / System CL)
+ *
+ * @param <A> logging-framework-specific Appender type
+ */
+public abstract class AbstractLogShadingHelper<A> {
+
+    private static final Object NULL_APPENDER = new Object();
+
+    private final ElasticApmTracer tracer;
+    private final LoggingConfiguration loggingConfiguration;
+    private final String serviceName;
+
+    public AbstractLogShadingHelper(ElasticApmTracer tracer) {
+        this.tracer = tracer;
+        loggingConfiguration = tracer.getConfig(LoggingConfiguration.class);
+        serviceName = tracer.getConfig(CoreConfiguration.class).getServiceName();
+    }
+
+    private static final WeakConcurrentMap<Object, Object> appenderToShadeAppender = new WeakConcurrentMap.WithInlinedExpunction<Object, Object>();
```

Comment:
Use the new `WeakMapSupplier.createMap()`. Also, maybe type to `<A, Object>`?

Reply:
Changed to the new sdk API.
As for typing - this is a static map, so I cannot type to an instance generic type